### ISSUE TITLE
[BE] Radical Independence

### ISSUE THREAD

**Issue by kenpaicat** [SloveniaEngineering/laguna-backend#43]
**This is a very low priority long distance issue ATM**.
Independence only applies to backend.

Get rid of following systems:
* [ ] crates.io by caching required crates in Alt Registry on GitHub
* [ ] GitHub
* [ ] Ubuntu APT manager in CI/CD

**Comment by kenpaicat** [comment]
> Getting rid of crates.io seems like a crazy requirement, even if we cache crates somewhere else, what about updates to those crates?

We just don't do version updates then. This app should work for some fixed dependency versions in case we can't get updates for whatever reason, that would mean having dependencies + `laguna-backend-*` in an alt registry (not crates.io).

> We can be independent from apt in CI/CD if we go back to builds without installing lld and clang, but I'm not sure we need independence from system package managers

Does the app build without lld, clang? Yes. We technically don't need them, they are there for linking speed.

> One thing is independence for running instances, the other is independence for development / developers.

Independance for instances is federative approach to torrent **tracking**, which I don't think we need, because it kinda already exists implicitly via "network" of trackers. Independence for development is a weird concept, maybe I misused the word independence.

Basically what I'm advocating for: if github is down, or ubuntu or alpine pacmans are down, or dockerhub is down: laguna should still work. Or to simplify: if we put laguna outside bunch of other systems it should still work.
